fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/TreeLike.java

```diff
@@ -210,24 +212,51 @@ default <V extends TreeLike<V>> Optional<V> mapMaybe(@Nonnull final BiFunction<T
         return Optional.ofNullable(foldNullable(Function.identity(), foldFunction));
     }
 
+    /**
+     * Method that maps the tree-like rooted at {@code this} to another tree-like of the same type. Unlike the other
+     * methods providing generic tree folds and maps, this method provides an easy way to mutate the leaves of the
+     * tree-like rooted at {@code this}, newly added leaves are passed again to the {@code replaceOperator}.
+     * @param replaceOperator an operator that translates a tree-like of {@code T} into another tree-like of type
+     *        {@code T}.
+     * @return an {@link Optional} of a tree-like object that is result of the tree-map operation if the fold of the
+     *         tree rooted at {@code this} exists, {@code Optional.empty()} otherwise
+     */
+    @Nonnull
+    default Optional<T> replaceLeavesMaybe(@Nonnull final UnaryOperator<T> replaceOperator) {
+        return replaceLeavesMaybe(replaceOperator, true);
```

Comment:
Shouldn't this be false? I felt better if the default would be false.

Reply:
done, plus some minor cleanup.